src/dg/strong_dg.cpp

```diff
@@ -31,6 +35,768 @@ DGStrong<dim,nstate,real,MeshType>::~DGStrong ()
+            Assert (current_cell->neighbor(iface).state() == dealii::IteratorState::valid, dealii::ExcInternalError());
+        
+//pcout<<"ON normal face"<<std::endl;
+            const auto neighbor_cell = current_cell->neighbor_or_periodic_neighbor(iface);
+            // Corresponding face of the neighbor.
+            // e.g. The 4th face of the current cell might correspond to the 3rd face of the neighbor
+            const unsigned int neighbor_iface = current_cell->neighbor_of_neighbor(iface);
+        
+            // Get information about neighbor cell
+            const unsigned int n_dofs_neigh_cell = this->operators.fe_collection_basis[neighbor_cell->active_fe_index()].n_dofs_per_cell();
+        
+            // Local rhs contribution from neighbor
+            std::vector<dealii::Tensor<1,dim,double>> neighbor_cell_rhs_aux (n_dofs_neigh_cell ); // Defaults to 0.0 initialization
+        
+            // Obtain the mapping from local dof indices to global dof indices for neighbor cell
+            neighbor_dofs_indices.resize(n_dofs_neigh_cell);
+            neighbor_cell->get_dof_indices (neighbor_dofs_indices);
+        
+            const auto metric_neighbor_cell = current_metric_cell->neighbor_or_periodic_neighbor(iface);
+            metric_neighbor_cell->get_dof_indices(neighbor_metric_dofs_indices);
+        
+            assemble_face_term_auxiliary (
+                iface, neighbor_iface, 
+                poly_degree, grid_degree,
+                current_dofs_indices, neighbor_dofs_indices,
+                current_metric_dofs_indices, neighbor_metric_dofs_indices,
+                current_cell_rhs_aux, neighbor_cell_rhs_aux);
+        
+            // Add local contribution from neighbor cell to global vector
+            for (unsigned int i=0; i<n_dofs_neigh_cell; ++i) {
+                for(int idim=0; idim<dim; idim++){
+                    rhs[idim][neighbor_dofs_indices[i]] += neighbor_cell_rhs_aux[i][idim];
+                }
+            }
+        } else {
+            // Should be faces where the neighbor cell has the same coarseness
+            // but will be evaluated when we visit the other cell.
+        }
+
+
+    } // end of face loop
+
+    // Add local contribution from current cell to global vector
+    for (unsigned int i=0; i<n_dofs_curr_cell; ++i) {
+        for(int idim=0; idim<dim; idim++){
+            rhs[idim][current_dofs_indices[i]] += current_cell_rhs_aux[i][idim];
+        }
+    }
+
+}
+template <int dim, int nstate, typename real, typename MeshType>
+void DGStrong<dim,nstate,real,MeshType>::assemble_auxiliary_residual ()
+{
+    using PDE_enum = Parameters::AllParameters::PartialDifferentialEquation;
+    using ODE_enum = Parameters::ODESolverParam::ODESolverEnum;
+    if ( (this->all_parameters->pde_type == PDE_enum::convection_diffusion || this->all_parameters->pde_type == PDE_enum::diffusion)
```

Comment:
So... strong form only works for convection_diffusion and diffusion??

Make things fail early on when they are constructed, or when parameters are read. Don't let code go through all this to skip the if-statement and fail later on because it would be missing the auxiliary terms, and access an out-of-bound address, which gives segfault and is super hard to debug.

Reply:
Strong form works for all physics. This is just for solving the auxiliary equations. If there is no diffusion then it skips doing the auxiliary solve and uses the values initialized at 0.

Reply:
We also have Navier Stokes and viscous Burgers. I'm guessing it hasn't been tested for nonlinear equations?

Reply:
This will be checked once there are strong form tests for those problems.

Reply:
OK, could you please make the code fail in that scenario then?

Reply:
fixed, I added it at this line in DG Strong.